Add standalone CLI options (#449)
biocore/empress: README.md

<i>"Empire plot" visualizing a phylogenetic tree of amplicon sequence variants (ASVs) in Empress, left, alongside a PCoA biplot in Emperor, right. As one of the ways in which these displays are integrated, selecting a tip in the tree (representing an ASV) enlarges the samples containing this ASV in Emperor -- thereby providing more information than would be available from either display alone.</i>
</div>

# Installation & Basic Usage

> Empress is available as either a standalone program or a QIIME 2 plugin. The standalone version will generate a folder with HTML/JS/CSS files necessary to view the plot while the QIIME 2 version will generate a `.qzv` Visualization that can be viewed on [https://view.qiime2.org/](https://view.qiime2.org/).

Reviewer: ```suggestion
Empress is available as either a standalone program or a QIIME 2 plugin. The standalone version will generate a folder with HTML/JS/CSS files necessary to view the plot, while the QIIME 2 version will generate a `.qzv` Visualization that can be viewed on [https://view.qiime2.org/](https://view.qiime2.org/) or by using `qiime tools view`.
```

## Standalone Version

EMPress is available through [PyPi](https://pypi.org/project/empress/). Run the following command to install Empress:

`pip install empress`

Try running the command `empress --help` to ensure that Empress has been installed properly. If you see details for the different Empress commands then your installation has completed and you are ready to start using Empress!.

> 

Reviewer: ```suggestion
### Available commands

Empress provides two commands: `empress tree-plot` and `empress community-plot`. Both commands generate an Empress visualization, but `community-plot` requires you to pass in a feature table and sample metadata while `tree-plot` only requires a tree file. See [this section](#first-a-note-about-empress-commands) of the docs for some more details.
```

feel free to adjust this as you see fit -- just trying to be clear about this

> ### Example standalone usage

Reviewer: ```suggestion
### Input files
```

> The standalone version of Empress takes the following filetypes as inputs.

Reviewer: ```suggestion
The standalone version of Empress takes the following filetypes as inputs. (Note that for `empress tree-plot` all of these except for the tree are optional, and for `empress community-plot` all except for the tree, feature table, and sample metadata are optional.)
```

| Input | Filetype |
| ----- | -------- |
| Tree | [Newick](https://en.wikipedia.org/wiki/Newick_format) |
| Feature Table | [biom](http://biom-format.org/) |
| Sample Metadata | TSV |
| Feature Metadata | TSV |
| PCoA | [skbio OrdinationResults](http://scikit-bio.org/docs/latest/generated/skbio.io.format.ordination.html) |

```

> empress community-plot \

Reviewer: This looks great. Could you add in a `tree-plot` example above this? Alternately, or some brief documentation that there are two commands available, to avoid users getting overwhelmed

The output will be a directory containing an `empress.html` file and a `support_files` directory containing the JS/CSS files required to view the plot in your browser. If you provided a PCoA to the `community-plot` command there will also be an `emperor-resources` subdirectory containing the files required to view the Emperor plot alongside the tree. You can view the `empress.html` file in any modern browser to interact with it the same way you would the QIIME 2 Visualization.

## QIIME 2 Version

See the [QIIME 2 installation](https://docs.qiime2.org/2020.8/install/) page for instructions on how to install QIIME 2. Once you have QIIME 2 installed, make sure the conda environment is activated by running:

`conda activate qiime2-2020.8`

> You can replace `qiime2-2020.8` above with whichever version of QIIME 2 you have currently installed. Install Empress through PyPi by using `pip install empress`. Once you have installed Empress, run the following commands to ensure that Empress was installed correctly. If you see information about Empress' QIIME 2 plugin, the installation was successful!

Reviewer: ```suggestion
You can replace `qiime2-2020.8` above with whichever version of QIIME 2 you have currently installed.

Now, run the following command to install Empress using PyPI:

`pip install empress`

Once you have installed Empress, run the following commands to ensure that Empress was installed correctly. If you see information about Empress' QIIME 2 plugin, the installation was successful!
```

This is really just adding a few line breaks to emphasize the `pip install empress`, so that impatient people like me don't skim through this and go "wait hold up why is QIIME saying that Empress isn't installed" ._.

```
qiime dev refresh-cache
qiime empress --help
```

> ### Example QIIME 2 usage

Reviewer: ```suggestion
### Available commands
Empress' QIIME 2 plugin provides two commands: `qiime empress tree-plot` and `qiime empress community-plot`. Both commands generate an Empress visualization, but `community-plot` requires you to pass in a feature table and sample metadata while `tree-plot` only requires a tree file. See [this section](#first-a-note-about-empress-commands) of the docs for some more details.

### Example QIIME 2 usage
```

I know this duplicates some of the text that I suggested above, but I can't think of a better way to do this, sorry ._. Feel free to push back on this -- if desired we could not bother with this change for right now. The main goal in adding this text is just preventing people from skimming, going "oh huh you need all those files for this??? i'm just gonna use another tool instead," and then ignoring empress for the rest of their lives :P

Reviewer: Actually thinking about it, I guess alternately maybe we could just add two commands for each `Example usage` section? (one showing tree plot, one showing community plot). That would also work -- I'll leave the decision up to you.

```
qiime empress community-plot \
    --i-tree tree.qza \
    --i-feature-table feature-table.qza \
    --m-sample-metadata-file sample_metadata.tsv \
    --m-feature-metadata-file taxonomy.qza \
    --i-pcoa ordination.qza \
    --p-filter-extra-samples \
    --o-visualization tree_viz.qzv
```

> # Tutorial: Using Empress in QIIME 2

Reviewer: I can't leave a suggestion at the correct line in the GitHub interface, but I would suggest changing the line that says "Empress currently has two commands available through QIIME 2" a little bit further down from here to remove the "through QIIME 2" part. This way we can refer users to this section from the standalone docs.

<i>"Empire plot" visualizing a phylogenetic tree of amplicon sequence variants (ASVs) in Empress, left, alongside a PCoA biplot in Emperor, right. As one of the ways in which these displays are integrated, selecting a tip in the tree (representing an ASV) enlarges the samples containing this ASV in Emperor -- thereby providing more information than would be available from either display alone.</i>
</div>

# Installation & Basic Usage

Empress is available as either a standalone program or a QIIME 2 plugin. The standalone version will generate a folder with HTML/JS/CSS files necessary to view the plot while the QIIME 2 version will generate a `.qzv` Visualization that can be viewed on [https://view.qiime2.org/](https://view.qiime2.org/) or by using `qiime tools view`.

## Standalone Version

EMPress is available through [PyPI](https://PyPI.org/project/empress/). Run the following command to install Empress:

`pip install empress`

> Try running the command `empress --help` to ensure that Empress has been installed properly. If you see details for the different Empress commands then the installation has succeeded and you are ready to start using Empress!.

Reviewer: ```suggestion
Try running the command `empress --help` to ensure that Empress has been installed properly. If you see details for the different Empress commands then the installation has succeeded and you are ready to start using Empress!
```

### Input files

The standalone version of Empress takes the following filetypes as inputs. (Note that for `empress tree-plot` all of these except for the tree are optional, and for `empress community-plot` all except for the tree, feature table, and sample metadata are optional.)

| Input | Filetype |
| ----- | -------- |
| Tree | [Newick](https://en.wikipedia.org/wiki/Newick_format) |
| Feature Table | [BIOM](http://biom-format.org/) |
| Sample Metadata | [TSV](https://en.wikipedia.org/wiki/Tab-separated_values) |
| Feature Metadata | [TSV](https://en.wikipedia.org/wiki/Tab-separated_values) |
| PCoA | [scikit-bio OrdinationResults](http://scikit-bio.org/docs/latest/generated/skbio.io.format.ordination.html) |

### Example standalone usage

```
empress tree-plot \
    --tree tree.nwk \
    --feature-metadata feature_metadata.tsv \
    --output-dir tree_viz
```

```
empress community-plot \
    --tree tree.nwk \
    --table feature-table.biom \
    --sample-metadata sample_metadata.tsv \
    --feature-metadata feature_metadata.tsv \
    --pcoa ordination.txt \
    --filter-extra-samples \
    --output-dir community_tree_viz
```

> You can view the details of the command line arguments with `empress tree-plot --help` and `empress community-plot --help`. Note that the path provided to `--output-dir` must not exist as it will be created by Empress upon successful execution of the command. It is also worth noting that the standalone version of the Empress commands does not support providing multiple sample/feature metadata files. If you have, for example, multiple feature metadata files, you should concatenate them all into one file that you pass to Empress.

Reviewer: ```suggestion
You can view the details of the command line arguments with `empress tree-plot --help` and `empress community-plot --help`. Note that the path provided to `--output-dir` must not exist as it will be created by Empress upon successful execution of the command. It is also worth noting that the standalone version of the Empress commands does not support providing multiple sample/feature metadata files. If you have, for example, multiple feature metadata files, you should combine them all into one file that you pass to Empress.
```

# Installation   

Empress is available as either a standalone program or a QIIME 2 plugin. The standalone version will generate a folder with HTML/JS/CSS files necessary to view the plot while the QIIME 2 version will generate a `.qzv` Visualization that can be viewed on [https://view.qiime2.org/](https://view.qiime2.org/).

## QIIME 2 Version

> If you are not using QIIME 2, feel free to skip to the next section.

Author: Cool - so maybe for the scope of this PR I can move the standalone example (and make the other requested changes). So then, if I'm understanding your comment correctly, the format of the README would look like this:

* Introduction
* QIIME2
    * Installation
    * Basic usage
* Standalone
    * Installation
    * Basic usage
* Comprehensive Q2 tutorial
* Pub, citation, acknowledgements

Then we can open up a new issue/discuss moving the current tutorial elsewhere. Is that in-line with what you were thinking?

> If you are not using QIIME 2, feel free to skip to the next section.

Reviewer: Yeah, that matches what I was thinking of. I would request we move the standalone section to above the QIIME 2 section, but aside from that this order of things sounds perfect 💯
